src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1366,32 +1366,12 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
     return;
   }
 
-  static size_t current_app = 0;
-  static size_t current_url = 0;
-  if (current_url >= urls.at(current_app).url.size()) {
-    ApplicationSharedPtr app;
-    current_url = 0;
-
-    bool is_default = false;
-    bool is_registered = false;
-    bool has_urls = false;
-    bool valid_app_found = false;
-    do {
-      if (++current_app >= urls.size()) {
-        current_app = 0;
-      }
-      const std::string& app_id = urls.at(current_app).app_id;
-      app = application_manager_.application_by_policy_id(app_id);
-
-      is_default = (app_id == policy::kDefaultId);
-      is_registered = (app && app->IsRegistered());
-      has_urls = !urls.at(current_app).url.empty();
-      valid_app_found = (is_default || (is_registered && has_urls));
-    } while (!valid_app_found);
+  AppIdURL app_url;
+  app_url = policy_manager_->GetNextUpdateUrl(urls);
+  while (!IsUrlAppIdValid(app_url.first)) {
+    app_url = policy_manager_->GetNextUpdateUrl(urls);
```

Comment:
As the Data Dictionary's Default section and its content (default urls) are mandatory, IsUrlAppIdValid will return true when app_url.first is "default".